deps/oblib/src/lib/oblog/ob_log_compressor.cpp

```diff
@@ -142,55 +172,44 @@ int ObLogCompressor::log_compress_block(char* dest, size_t dest_size, const char
   return ret;
 }
 
-void * ObLogCompressor::log_compress_thread(void *arg)
-{
-  if (NULL == arg) {
-    LOG_STDERR("log_compress_thread init failed due to invalid arguments.\n");
-  } else {
-    ObLogCompressor * compressor = reinterpret_cast<ObLogCompressor *>(arg);
-    prctl(PR_SET_NAME, "syslog_compress");
-    compressor->log_compress();
-  }
-  return NULL;
-}
-
 void ObLogCompressor::log_compress()
 {
-  static int sleep_us = 100 * 1000; // 100ms
-  int src_size   = DEFAULT_COMPRESSION_BLOCK_SIZE;
-  int dest_size  = DEFAULT_COMPRESSION_BUFFER_SIZE;
-  char *src_buf  = (char *)malloc(src_size);
-  char *dest_buf = (char *)malloc(dest_size);
+  int ret = OB_SUCCESS;
+  static int sleep_us = 100 * 1000;  // 100ms
+  int src_size = DEFAULT_COMPRESSION_BLOCK_SIZE;
+  int dest_size = DEFAULT_COMPRESSION_BUFFER_SIZE;
+  char *src_buf = (char *)ob_malloc(src_size);
+  char *dest_buf = (char *)ob_malloc(dest_size);
   if (!src_buf || !dest_buf) {
-    LOG_STDERR("Failed to malloc.\n");
-  } else while (!has_stoped_) {
-    std::string file_name;
-    pthread_mutex_lock(&log_compress_mutex_);
-    if (file_list_.empty()) {
-      pthread_cond_wait(&log_compress_cond_, &log_compress_mutex_);
-    }
-    if (!has_stoped_ && !file_list_.empty()) {
-      file_name = file_list_.front();
-      file_list_.pop_front();
-    }
-    pthread_mutex_unlock(&log_compress_mutex_);
+    LOG_STDERR("Failed to ob_malloc.\n");
+  } else {
+    while (!has_stoped_) {
+      ObString *file_name = NULL;
+      log_compress_cond_.lock();
+      while (0 >= file_list_.get_total() && !has_stoped_) {
+        log_compress_cond_.wait(0);
+      }
+      if (!has_stoped_) {
+        ret = file_list_.pop(file_name);
+      }
+      log_compress_cond_.unlock();
 
-    if (has_stoped_ || file_name.empty() || 0 != access(file_name.c_str(), F_OK)) {
-    } else {
-      std::string compression_file_name = get_compression_file_name(file_name);
-      FILE *input_file = NULL;
-      FILE *output_file = NULL;
-      if (NULL == (input_file = fopen(file_name.c_str(), "r"))) {
-        LOG_STDERR("Fialed to fopen, err_code=%d.\n", errno);
-      } else if (NULL == (output_file = fopen(compression_file_name.c_str(), "w"))) {
-        fclose(input_file);
-        LOG_STDERR("Fialed to fopen, err_code=%d.\n", errno);
+      if (has_stoped_ || NULL == file_name || file_name->empty() || 0 != access(file_name->ptr(), F_OK)) {
```

Comment:
access accept c_str with '\0' tailer, but file_name created before is not expected